File: react/src/lib/domain/entities/DataObjectLocationType.ts
import BaseEnum from './BaseEnum';

export default class DataObjectLocationType extends BaseEnum {
    public static MY_COMPUTER = new DataObjectLocationType('1', 'dataObject:dataObjectLocationTypeEnum:myComputer');
    public static DCA_SERVER = new DataObjectLocationType('2', 'dataObject:dataObjectLocationTypeEnum:dcaServer');
    public static FTP = new DataObjectLocationType('3', 'dataObject:dataObjectLocationTypeEnum:ftp');

    public constructor(id?: string, name?: string) {
        super(id, name);
    }
}

File: backend/src/main/java/com/equisoft/dca/backend/dataobject/model/DataObjectLocationType.java
package com.equisoft.dca.backend.dataobject.model;

import com.equisoft.dca.backend.common.model.BaseEnum;

import lombok.Getter;

@Getter
public enum DataObjectLocationType implements BaseEnum<Integer> {

	MY_COMPUTER(1, "My computer"),
	DCA_SERVER(2, "DCA server"),
	FTP(3, "FTP");

	private final int id;

	private final String name;

	DataObjectLocationType(int id, String name) {
		this.id = id;
		this.name = name;
	}

	@Override
	public Integer getId() {
		return id;
	}
}
